### DISCUSSION TITLE
Warning about infinite loops

### DISCUSSION THREAD

**Discussion by rowanseymour** [nyaruka/floweditor#1121] [Ideas]
We allow these now, but maybe they still warrant some kind of visual warning to the user?

Have a ticket where the user inadvertently created a loop.

**Comment by nicpottier** [comment]
Just as a note I ran into this as well when playing with intents and was totally confused. I feel we should definitely show something to say why we aren't connecting here.

**Comment by rowanseymour** [comment]
Are we talking about two different things? You weren't able to make a loop but this user had inadvertently created a loop... in what case can't you create a loop?

**Comment by nicpottier** [comment]
Oh, it didn't connect for me, but no warning. Thought that was the issue.

**Comment by rowanseymour** [comment]
Ok from playing around.. you can't join an action to an action to the first action.. but you can if there's a split, even a passive one, in the loop... is that the rule? Seems like both cases might benefit from some visual explanation

**Comment by nicpottier** [comment]
I ran across this when having an intent split without a wait, so may not be as simple as any split.

**Comment by rowanseymour** [comment]
Seems like the rule could be that _there needs to be a split,_ since any split provides a mechanism by which the loop could end.

i.e. just check the graph for closed loops
